app/javascript/src/components/dashboard/MyShifts.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import moment from 'moment';
+
+import { Card, Header } from 'semantic-ui-react';
+
+class MyShifts extends Component {
+
+    constructor(props) {
+        super(props);
+    }
+
+    filterShifts = (shift) => {
+        // NOTE(anesu): We only want shifts from the next 7 days
+        const daysToAdd = 7;
+
+        const shiftStartTime = new Date(shift.start);
+        const now = new Date();
+        const sevenDaysFromNow = new Date();
+        sevenDaysFromNow.setDate(now.getDate() + daysToAdd); 
+
+
+        return shiftStartTime > now && shiftStartTime < sevenDaysFromNow;
+    }
+
+    sortByEarliestShiftsFirst = (shift1, shift2)  => {
+        return new Date(shift1.start) > new Date(shift2.start);
+    }
+
+    mapShiftToCard = (shift) => {
+        const start = moment(new Date(shift.start))
+            .calendar();
+        const end = moment(new Date(shift.end))
```

Comment:
what does this formatting do?

Reply:
It's taking the user shifts and turning them into a UI element to display in the card